src/webots/nodes/WbMotor.cpp

```diff
@@ -127,20 +130,31 @@ void WbMotor::postFinalize() {
   if (!mMuscles->isEmpty() || robot()->maxEnergy() > 0)
     setupJointFeedback();
 
+  inferMotorCouplings();
+  checkMultiplierAcrossCoupledMotors();  // it calls all the other checks implicitly
+
+  // must be done in the postFinalize step as the coupled motor consistency check could change the values
+  mTargetVelocity = mMaxVelocity->value();
+  mMotorForceOrTorque = mMaxForceOrTorque->value();
+
+  mNeedToConfigure = true;  // notify libController about the changes done by the check
+
   WbMFIterator<WbMFNode, WbNode *> it(mMuscles);
   while (it.hasNext())
     dynamic_cast<WbMuscle *>(it.next())->postFinalize();
 
-  connect(mMaxVelocity, &WbSFDouble::changed, this, &WbMotor::updateMaxVelocity);
-  connect(mAcceleration, &WbSFDouble::changed, this, &WbMotor::updateMaxAcceleration);
+  connect(mMultiplier, SIGNAL(changed()), this, SLOT(updateMultiplier()));
+  connect(mMaxVelocity, SIGNAL(changed()), this, SLOT(updateMaxVelocity()));
+  connect(mMinPosition, SIGNAL(changed()), this, SLOT(updateMinAndMaxPosition()));
+  connect(mMaxPosition, SIGNAL(changed()), this, SLOT(updateMinAndMaxPosition()));
+  connect(mAcceleration, SIGNAL(changed()), this, SLOT(updateMaxAcceleration()));
+  connect(mMaxForceOrTorque, SIGNAL(changed()), this, SLOT(updateMaxForceOrTorque()));
```

Comment:
The reason I had to change it to this was because the slot functions had default values which isn't otherwise supported in the new syntax. Either way, I can do it without having to rely on default parameters so I changed it back to what it was.